docs/source/extensions/callbacks.rst

```diff
@@ -422,12 +422,24 @@ on_exception
 .. automethod:: pytorch_lightning.callbacks.Callback.on_exception
     :noindex:
 
+state_dict
+~~~~~~~~~~~~~~~~~~
+
+.. automethod:: pytorch_lightning.callbacks.Callback.state_dict
+    :noindex:
+
 on_save_checkpoint
```

Comment:
same here. we should remove the explicit mention to reduce the visibility. From an API point of view, they are already present in the API references.

Reply:
Thinking out loud: Should we still mention but also say that this is deprecated?

Reply:
anyway is better. I usually remove it since if it's a new user, they better not know about this and if it's an old user, probably they will search for it and API references will pop up for them.

Reply:
(see above [comment](https://github.com/PyTorchLightning/pytorch-lightning/pull/12351#discussion_r829276479))